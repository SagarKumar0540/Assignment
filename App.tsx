import AppNavigator from "./src/navigation/appNavigator";
import { Provider } from "react-redux";
import { store } from "./src/state/store";
import { ThemeProvider } from "./src/context/themeContext";

const App = () => {
  return (
    <ThemeProvider>
      <Provider store={store} >
        <AppNavigator />
      </Provider>
    </ThemeProvider>
  )
}


export default App;