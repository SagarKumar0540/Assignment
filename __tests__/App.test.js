// __tests__/App.test.js
import React from 'react';
import { render } from '@testing-library/react-native';
import App from '../App';

// Mock the AppNavigator component to avoid rendering actual navigation components
jest.mock('../src/navigation/appNavigator', () => () => (
  <></>
));

// Mock the Redux store provider to avoid test issues with actual store
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  Provider: ({ children }) => <>{children}</>,
}));

// Mock the ThemeProvider to provide basic context
jest.mock('../src/context/themeContext', () => ({
  ThemeProvider: ({ children }) => <>{children}</>,
}));

describe('App', () => {
  it('renders without crashing and matches snapshot', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});