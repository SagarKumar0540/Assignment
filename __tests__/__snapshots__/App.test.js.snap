// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing and matches snapshot 1`] = `null`;
